### PR TITLE
Automatically remove excess filler if there's too much

### PR DESCRIPTION
**Pull request by silasary**
This is no substitute for making a properly sized world.  But it's better than what we're currently doing

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -331,13 +335,33 @@ def add_filler_items(self, item_pool, traps):
...
+            for _ in range(0, trap_count):
                 extra_item = self.create_item(self.random.choice(traps))
                 item_pool.append(extra_item)
 
-            for i in range(0, filler_count):
+            for _ in range(0, filler_count):
                 extra_item = self.create_item(filler_item_name)
                 item_pool.append(extra_item)
+        elif extras < 0:
+            print(f"Warning: {self.game} has more items than locations. {abs(extras)} useless items will be removed at random.")
```
Comment (nicopop) [inline src/__init__.py:346]: maybe use a logger instead of print so it shows in log?

Comment (silasary) [reply src/__init__.py:346]: Oh, yes. I definitely meant to do that.

File: src/__init__.py [hunk 2]
```diff
@@ -331,13 +336,33 @@ def add_filler_items(self, item_pool, traps):
...
+            for _ in range(0, abs(extras)):
+                popped = None
+                if fillers:
+                    popped = fillers.pop()
+                elif traps:
+                    popped = traps.pop()
+                elif useful:
+                    popped = useful.pop()
+                else:
+                    print("Warning: Could not remove enough non-progression items from the pool.")
```
Comment (nicopop) [inline src/__init__.py:363]: this one too ;)

**Comment by nicopop** [comment]
dont forget to change the call to the self.add_filler_items at line 172

**Review by nicopop — CHANGES_REQUESTED** [review]
Since you deprecated add_filler_items you also need to change line 172 of \_\_init__.py or it will get called when generating.

**Comment by silasary** [comment]
Did I seriously forget to stage that line?

**Review by nicopop — DISMISSED** [review]
Looks good now 👍

File: src/__init__.py [hunk 3]
```diff
@@ -331,13 +336,33 @@ def add_filler_items(self, item_pool, traps):
...
+            for _ in range(0, trap_count):
                 extra_item = self.create_item(self.random.choice(traps))
                 item_pool.append(extra_item)
 
-            for i in range(0, filler_count):
+            for _ in range(0, filler_count):
                 extra_item = self.create_item(filler_item_name)
                 item_pool.append(extra_item)
+        elif extras < 0:
+            logging.warning(f"{self.game} has more items than locations. {abs(extras)} useless items will be removed at random.")
```
Comment (FuzzyGamesOn) [inline src/__init__.py:347]: I think calling the items useless when useful is included could be misleading. We should change this to "non-progression".

**Review by FuzzyGamesOn — CHANGES_REQUESTED** [review]
Just a minor verbiage change for clarity 😄

**Review by FuzzyGamesOn — APPROVED** [review]

**Pull request merged**